sdks/python/apache_beam/io/range_trackers.py

```diff
@@ -324,3 +324,123 @@ def set_current_position(self, position):
 
   def fraction_consumed(self):
     return self._range_tracker.fraction_consumed()
+
+
+class LexicographicKeyRangeTracker(iobase.RangeTracker):
+  """
+  A range tracker that tracks progress through a lexicographically
+  ordered keyspace of strings.
+  """
+
+  def __init__(self, start_key=None, stop_key=None):
+    self._start_key = start_key
+    self._stop_key = stop_key
+    self._lock = threading.Lock()
+    self._last_claim = None
+
+  def start_position(self):
+    return self._start_key
+
+  def stop_position(self):
+    return self._stop_key
+
+  def try_claim(self, position):  # pylint: disable=unused-argument
```

Comment:
* Why disable=unused-argument? Both arguments are used.
* Perform the same set of sanity checks as the Java ByteKeyRangeTracker https://github.com/apache/incubator-beam/blob/master/sdks/java/core/src/main/java/org/apache/beam/sdk/io/range/ByteKeyRangeTracker.java - applies to all other methods too